utils/apply-tag.ts

```diff
@@ -0,0 +1,32 @@
+import { IAspect, ITaggable, TagManager } from 'aws-cdk-lib';
+import { IConstruct } from 'constructs';
+
+export type Tags = { [key: string]: string } & {
+    environment: 'development' | 'staging' | 'production' | 'demonstration';
+    project: string;
+    owner: string;
+};
+
+export class ApplyTags implements IAspect {
+  #tags: Tags;
+
+  constructor(tags: Tags) {
+    this.#tags = tags;
+  }
+
+  visit(node: IConstruct) {
+    if (TagManager.isTaggable(node)) {
+      Object.entries(this.#tags).forEach(([key, value]) => {
+        console.log(`Applying tag ${key}=${value} to ${node.node.path}`);
+        this.applyTag(node, key, value);
```

Comment:
**Suggestion:** The `console.log` statement inside the `visit` method is not necessary for production code and can lead to performance issues and cluttered logs. It is better to remove it or use a logging library with appropriate log levels.
> 🏺  **Medium Priority** 
 > [category: **`maintainability`**, importance: **`5`**]
> **Reason:** Removing unnecessary console logs can help in maintaining cleaner code and reducing log clutter, which is good for production environments.
 

```suggestion
        this.applyTag(node, key, value);
```